test/SAM/README.md

```diff
@@ -0,0 +1,153 @@
+	beyond the end of a reference we must store the bases verbatim
+	and not read past the end of the supplied reference.
+
+c1#clip.sam
+	Permutations of soft and hard clipping
+
+c1#noseq.sam
+c1#unknown.sam
+	Alignments without any sequence or quality strings; with and
+	without MD and NM tags.
+
+	Intent: validate CRAM can store MD / NM tags as the data
+	necessary to recreate them is absent.
+
+c1#pad1.sam
+c1#pad2.sam
+c1#pad3.sam
+c2#pad.sam
+	Various combinations of the CIGAR "P" operator in conjunction
+	with insertions and deletions, including adjacent I/D codes.
+
+	Intent: test the ability of parsing CIGAR, but also may be
+	useful as a test for "pileup" algorithms.
+
+	Basic simple alignments.
+
+ce#1.sam
+ce#2.sam
+ce#5.sam
+ce#1000.sam
+	Basic simple alignments.  Tests input consisting of just 1, 2,
+	5 and 1000 sequences.
+
+ce#5b.sam
+	Basic simple alignments.
+	Tests multiple chromosomes.
+
+ce#large_seq.sam
+	Stress test for long CIGARs.
+
+ce#supp.sam
+	Contains SUPPLEMENTARY alignments
+
+ce#tag_depadded.sam
+ce#tag_padded.sam
+	Contains consensus tags (CT) and read tags (PT) auxiliary
+	tags, for testing section 3 of the SAM spec.
+
+ce#unmap.sam
+	Contains entirely unmapped data, with no SAM header.
+
+ce#unmap1.sam
+	Contains entirely unmapped data, but has a SAM header.
+
+ce#unmap2.sam
+	A mix of mapped and unmapped data
+
+fieldarith.sam
+	Testing of SAM parsing, with tags to describe the reference
+	and query lengths and alignment end position.
+
+md#1.sam (remove?)
+	Stress test for MD auto-generation, including neighbouring
+	SNPs (0 distance between them) and indels.
+
+	Intent: CRAM validation.
+	(CHECK: are all valid, or do we also use this for validating
+	round-trip while preserving incorrect MD tags?)
+
+xx#MD.sam
+	All MD and NM values are correctly computed
+
+xx#MD2.sam
+	All MD and NM values are incorrect computed.
+	Intent: validate CRAM round-trip correctly preserves invalid
+	data without fixing it by auto-generation.
+
+xx#blank.sam
+	No alignments or headers other than a comment.
+
+xx#large_aux.sam
+	Large numbers of auxiliary tags and/or single very large aux tags.
+
+xx#large_aux2.sam
+	Lots of combinations of different auxiliary tags.
+	Intent: test the CRAM tag dictionary.
+
+xx#minimal.sam
+	Zero length sequences.  These are combined with valid, but
+	non-helpful cigar strings, such as "10H" or "5H0M5H".
+
+xx#pair.sam
+	Basic read-pair tests
+
+xx#repeated.sam
+	The same pair repeated multiple times (not position sorted)
+
+xx#rg.sam
+	Testing of multiple read-groups.
+
+xx#tlen.sam
+xx#tlen2.sam
+	Testing of the two possible interpretations of TLEN:
+	leftmost to rightmost (spec compliant) and 5' to 3' (most
+	commonly used).
+
+	Intent: CRAM round-trips shouldn't change TLEN even when data
+	is not spec compilant.
+
+xx#triplet.sam
+	A template with 3 components (READ1, READ2 and neither).
+
+xx#unsorted.sam
+	Neither name nor chr / position sorted.
+
+
+index.sam	?
```

Comment:
1. should the bams and crams include an index so that implementations can check that they can all read & use one agreed-upon index implementation?
1. in a similar vein, since we seem to have many different index naming scenes allowed, should we have some tests for that:
* file.bam + file.bai 
* file.bam + file.bam.bai
* file.sam + file.sai ?
* file.sam + file.sam.sai?
* ....

3. what about search order, i.e. in the presence of two indexes, file.bam.bai & file.bai, do we want to specify which index should be used?